### PR TITLE
More fixes for issue #3925

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

**Comment by stweil** [comment]
With these fixes the test for issue #3925 runs further, but still fails finally:

```
+ tesseract num.ocra.exp0.png num.ocra.exp0 nobatch box.train
APPLY_BOXES:
   Boxes read from boxfile:      10
   Found 10 good blobs.
Generated training data for 1 words
+ unicharset_extractor num.ocra.exp0.box
Extracting unicharset from box file num.ocra.exp0.box
Wrote unicharset file unicharset
+ set_unicharset_properties -U unicharset -O num.unicharset --script_dir=/home/stweil/src/github/tesseract-ocr/langdata/
Loaded unicharset of size 13 from file unicharset
Setting unichar properties
Setting script properties
Writing unicharset to file num.unicharset
+ shapeclustering -F font_properties -U num.unicharset num.ocra.exp0.tr
Reading num.ocra.exp0.tr ...
Building master shape table
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0 1 2 3 4 5 6 7 8 9
Stopped with 0 merged, min dist 0.256098
Master shape_table:Number of shapes = 10 max unichars = 1 number with multiple unichars = 0
+ mftraining -F font_properties -U num.unicharset -O num.unicharset num.ocra.exp0.tr
Read shape table shapetable of 10 shapes
Reading num.ocra.exp0.tr ...
Warning: no protos/configs for Joined in CreateIntTemplates()
Warning: no protos/configs for |Broken|0|1 in CreateIntTemplates()
Class->NumConfigs == this->fontset_table_.at(Class->font_set_id).size():Error:Assert failed:in file ../../../src/classify/intproto.cpp, line 950
Illegal instruction
```

**Comment by stweil** [comment]
@SpaceView, @zdenop, could you please try whether you still have an issue after applying this pull request? Here is my test result:

```
+ tesseract num.ocra.exp0.png num.ocra.exp0 nobatch box.train
APPLY_BOXES:
   Boxes read from boxfile:      10
   Found 10 good blobs.
Generated training data for 1 words
+ unicharset_extractor num.ocra.exp0.box
Extracting unicharset from box file num.ocra.exp0.box
Wrote unicharset file unicharset
+ set_unicharset_properties -U unicharset -O num.unicharset --script_dir=/home/stweil/src/github/tesseract-ocr/langdata/
Loaded unicharset of size 13 from file unicharset
Setting unichar properties
Setting script properties
Writing unicharset to file num.unicharset
+ shapeclustering -F font_properties -U num.unicharset num.ocra.exp0.tr
Reading num.ocra.exp0.tr ...
Building master shape table
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances...
Stopped with 0 merged, min dist 999.000000
Computing shape distances... 0 1 2 3 4 5 6 7 8 9
Stopped with 0 merged, min dist 0.256098
Master shape_table:Number of shapes = 10 max unichars = 1 number with multiple unichars = 0
+ mftraining -F font_properties -U num.unicharset -O num.unicharset num.ocra.exp0.tr
Read shape table shapetable of 10 shapes
Reading num.ocra.exp0.tr ...
Warning: no protos/configs for Joined in CreateIntTemplates()
Warning: no protos/configs for |Broken|0|1 in CreateIntTemplates()
Done!
+ cntraining num.ocra.exp0.tr
Reading num.ocra.exp0.tr ...
Clustering ...

Writing normproto ...
+ mv inttemp num.inttemp
+ mv pffmtable num.pffmtable
+ mv normproto num.normproto
+ mv shapetable num.shapetable
+ combine_tessdata num.
Version:5.2.0-108-gf77c
1:unicharset:size=837, offset=192
3:inttemp:size=134553, offset=1029
4:pffmtable:size=110, offset=135582
5:normproto:size=1382, offset=135692
13:shapetable:size=184, offset=137074
23:version:size=15, offset=137258
Combining tessdata files
Output num.traineddata created successfully.
+ mkdir tessdata
+ mv num.traineddata tessdata
+ tesseract num.ocra.exp0.png - --psm 7 -l num --tessdata-dir tessdata
0123456789
```

File: src/classify/intproto.cpp [hunk 1]
```diff
@@ -505,9 +505,11 @@ INT_TEMPLATES_STRUCT *Classify::CreateIntTemplates(CLASSES FloatProtos,
...
     assert(UnusedClassIdIn(IntTemplates, ClassId));
     IClass = new INT_CLASS_STRUCT(FClass->NumProtos, FClass->NumConfigs);
-    FontSet fs{FClass->font_set.size()};
-    for (unsigned i = 0; i < fs.size(); ++i) {
-      fs[i] = FClass->font_set.at(i);
+    unsigned fs_size = FClass->font_set.size();
+    FontSet fs;
+    fs.reserve(fs_size);
+    for (unsigned i = 0; i < fs_size; ++i) {
+      fs.push_back(FClass->font_set.at(i));
```
Comment (egorpugin) [inline src/classify/intproto.cpp:512]: no need to `at()`, you just got `FClass->font_set.size();`

Comment (stweil) [reply src/classify/intproto.cpp]: Thanks. This is fixed now.

**Review by zdenop — APPROVED** [review]

**Pull request merged**